data/Record.js

```diff
@@ -118,6 +120,24 @@ export class Record {
         return this.store.getAncestorsById(this.id, false);
     }
 
+    /** @return {boolean} */
+    @computed
```

Comment:
think the use of computed breaks the immutability here, it implies a cached value that will change..